src/main/java/mytown/util/TownBlockKey.java

```diff
@@ -10,7 +10,7 @@ public TownBlockKey(int dim, int x, int z) {
 
     @Override
     public int hashCode() {
-        return this.dim ^ this.x ^ this.z;
+        return (this.dim + 13121) * (this.x | 256) * (this.z | 256);
```

Comment:
what does this change?
why not clasic
```
        int result = dim;
        result = 31 * result + x;
        result = 31 * result + z;
        return result;
```
or Objects.equals?

Reply:
is this for dim == 0?

Reply:
Yeah, sure, that would probably be fine too. I'll swap to that.
The reason for the change is that xor produces the same hash for values that are close to each other, which is a problem, because the data set is chunks claimed by players, so claimed chunks being near each other is very common.